orttraining/orttraining/core/graph/gradient_builder.cc

```diff
@@ -696,15 +696,8 @@ IMPLEMENT_GRADIENT_BUILDER(GetConvGradient) {
 IMPLEMENT_GRADIENT_BUILDER(GetSigmoidGradient) {
   auto const_one = OneConstantNode(OElemType(0));
   return std::vector<NodeDef>{
-      const_one,
-      NodeDef("Sub",
-              {const_one.output_args[0], O(0)},
-              {IA("one_minus_output")}),
-      NodeDef("Mul",
-              {IA("one_minus_output"), O(0)},
-              {IA("sigmoid_derivate")}),
-      NodeDef("Mul",
-              {IA("sigmoid_derivate"), GO(0)},
+      NodeDef(OpDef{"SigmoidGrad", kMSDomain, 1},
+              {GO(0), I(0)},
```

Comment:
We should take O(0) here, so that we don't need to recomputed sigmoid in the gradient kernel. 

dx = dy * O(0) * (1- O(0))